Gene view redesign

mdml opened the pull request:
- Rewrite save to use librsvg to convert to PDF/PNG/SVG
- Simplify printing greatly using "data-" attributes in links
- Redesign of the five views, putting them in panels and adding common 
- Use grid layout for views/control panel, and fix control panel in place using Bootstrap's affix
- Add icons to control toggling, saving, printing, instructions, and (eventually) tours

Still to-do: implement tours for "?" icons

Commit:
initial version of main view redesign

Commit:
fix some formatting issues with the control panel

Commit:
rewrite save functionality

Commit:
Add librsvg to dependencies

Commit:
update navbar and remove old download code

Commit:
Merge branch 'view-redesign' of https://github.com/raphael-group/magi into view-redesign

Commit:
small improvements to layout and visuals

Commit:
small improvements in how the dataset panel is created

Commit:
improve UX of control panel

johndashen commented:
The panels on the main gene view page as a whole look great.  However, I can't see the links in the upper right corner of each panel.  

A few notes:
- In the aberrations view, the legend mouseover is cut off by the panel boundary.
- Also in the aberrations view, the Sort columns[+] text appears twice, once on the left and once on the right.
- The Control Panel > Share link doesn't display a link.

I like how the control panel stays on the same position on the page, which is a nice effect. I wonder if we could also keep the navbar on top of the page while letting only the gene view panels scroll, and do this for other pages in MAGI as well.

johndashen commented:
Also, this may be from another branch, but on the front page, the example queries (SWI-SNF/TCGA Pancancer, etc.) no longer change the chosen datasets in the dropdown.

johndashen commented on views/view.jade:
```diff
@@ -5,233 +5,255 @@ block body
+                    | Copy Number Aberrations
+                  ul(class="list-inline pull-right")
+                    li
+                      a(class="fa fa-info", title="CNA view description", tabindex="0", data-toggle="popover", data-trigger="hover", data-html="true", data-placement="bottom", data-content="Shows the copy number aberrations surrounding the selected gene (highlighted as red) on the chromosome. The topest panel shows genes (shown as blocks) distributed in the region. You can click and drag a window ...
```
topest -> topmost

johndashen commented on views/view.jade:
```diff
@@ -5,233 +5,255 @@ block body
+                    | Transcript
+                  ul(class="list-inline pull-right")
+                    li
+                      a(class="fa fa-info", title="Transcript view description", tabindex="0", data-toggle="popover", data-trigger="hover", data-html="true", data-placement="bottom", data-content="Shows the mutations in each REFSEQ transcript of each gene in the subnetwork. Mutations are colored by cancer type. Each symbol represents a distinct mutation type. Putatively inactivating mutations are ...
```
The description refers selectors for protein domain databases (Pfam, SMART, etc.).  Are they visible somewhere?

Commit:
move sample sorting into an aberration matrix icon dropdown. fix missing share link.

Commit:
fix typos in instructions

mdml commented:
Good catches on the typos, I cleared them up.

I also fixed the other three issues you mentioned. I moved sorting into a icon in the aberrations panel. You need to run bower install to install the new jquery-ui dependency, and also need to clone the latest version of gd3.

johndashen commented:
Looks good now.  I'll merge this into the link-django-annos branch and put
it up on the node MAGI machines.

On Thu, Jan 7, 2016 at 3:05 PM, Max Leiserson notifications@github.com
wrote:

> Good catches on the typos, I cleared them up.
> 
> I also fixed the other three issues you mentioned. I moved sorting into a
> icon in the aberrations panel. You need to run bower install to install the
> new jquery-ui dependency, and also need to clone the latest version of gd3.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/raphael-group/magi/pull/27#issuecomment-169791084.
